app.py

```diff
@@ -209,7 +209,34 @@ def _get_app_path(self, version=None):
 
     def _launch_app(self, context, file_to_open=None, version=None):
         """
-        Launches an app
+        Launches an application. No environment variable change is leaked to the outside world.
+        :param context: Tank context we will opening the app in.
+        :param file_to_open: Optional file to open when the app launches. Can be None.
+        :param version: Version of the app to launch. Specifying None means the default version will
+                        be picked.
+        """
+        try:
+            # Clone the environment variables
+            environ_clone = os.environ.copy()
+            sys_path_clone = sys.path[:]
+            self.__launch_app_internal(context, file_to_open, version)
+        finally:
+            # Clear the original structures and add into them so that users who did
+            # from os import environ and from sys import path get the restored values.
+            os.environ.clear()
+            for k, v in environ_clone.items():
+                os.environ[k] = v
+            del sys.path[:]
+            sys.path.extend(sys_path_clone)
+
+    def __launch_app_internal(self, context, file_to_open=None, version=None):
```

Comment:
prefer single underscore at the beginning of the method name (actually I prefer double but I think we decided that we should avoid using it!)